pkg/base1/test-file.js

```diff
@@ -483,6 +483,50 @@ QUnit.test("watching without reading", function (assert) {
     }
 });
 
+QUnit.test.skipWithPybridge("watching directory", assert => {
+    const done = assert.async();
+    assert.expect(15);
+
+    let n = 0;
+    const watch = cockpit.channel({ payload: "fswatch1", path: dir });
+    watch.addEventListener("message", (event, payload) => {
+        const msg = JSON.parse(payload);
+        n += 1;
+
+        if (n == 1) {
+            assert.equal(msg.event, "created", "world.txt created");
+            assert.equal(msg.path, dir + "/world.txt");
+            assert.notEqual(msg.tag, "-");
+        } else if (n == 2) {
+            assert.equal(msg.event, "changed", "world.txt changed");
+            assert.equal(msg.path, dir + "/world.txt");
+            assert.notEqual(msg.tag, "-");
+        } else if (n == 3) {
+            assert.equal(msg.event, "done-hint", "world.txt done-hint");
+            assert.equal(msg.path, dir + "/world.txt");
+            assert.notEqual(msg.tag, "-");
+
+            cockpit.spawn(["sh", "-c", `chmod 001 ${dir}/world.txt`]);
+        } else if (n == 4) {
+            assert.equal(msg.event, "attribute-changed", "world.txt attribute-changed");
+            assert.equal(msg.path, dir + "/world.txt");
+            assert.notEqual(msg.tag, "-");
+
+            cockpit.spawn(["sh", "-c", `rm ${dir}/world.txt`]);
```

Comment:
Note: The protocol defines `moved`, but at least `mv world.txt planet.txt` just looks like a `deleted` plus another `created` on my system. That feels file system specific, so I didn't include it in the unit test.